### PR TITLE
Game logic

### PR DESCRIPTION
**Pull request by geraldgsh**
- Created logic to find winner and draw

### CODE REVIEW

File: bin/main.rb [hunk 1]
```diff
@@ -1,37 +1,52 @@
 #!/usr/bin/env ruby
 require '../lib/engine.rb'
```
Comment (mcrd25) [inline bin/main.rb:2]: [OPTIONAL]
It is convention to use `require_relative` for file in the directory of the project and `require` for external files like gems etc.

File: lib/engine.rb [hunk 2]
```diff
@@ -1,64 +1,130 @@
...
+      arr.each do |elem|
+        print elem.state
+        print ' '
+      end
+      puts "\n"
     end
   end
 end
+
+class InputManipulator
```
Comment (mcrd25) [inline lib/engine.rb:108]: [OPTIONAL]
Maybe try putting this class in a separate file as it is better separation of code.
If you really want to have it in this engine file you can always require it

File: lib/engine.rb [hunk 3]
```diff
@@ -1,64 +1,130 @@
...
   end
 
-  # Method to mark O on board
   def make_o
     @state = 'O'
   end
 end
 
-# Tic Tack Toe grid
 class Board
```
Comment (mcrd25) [inline lib/engine.rb:72]: [OPTIONAL]
Maybe try putting this class in a separate file as it is better separation of code.
If you really want to have it in this engine file you can always require it

File: lib/engine.rb [hunk 4]
```diff
@@ -1,64 +1,130 @@
 # frozen_string_literal: true
 
-# Class for game state
 class Game
```
Comment (mcrd25) [inline lib/engine.rb:3]: [OPTIONAL]
Maybe try putting this class in a separate file as it is better separation of code.
If you really want to have it in this engine file you can always require it

**Review by mcrd25 — CHANGES_REQUESTED** [review]
# Changes Required. Let's get this done 💪 

Hi, @geraldgsh and @webmarkyn 👋,

Great attempt so far!
I must say your main.rb is really good! You are the first pair I've seen actually implement classes to handle the input you get from main.rb as opposed to simply transferring all the methods that have `chomps` and `puts` statements to main.rb. I really applaud this 👏 👏 👏. Made my day! 😄 

**This being said, there is one issue for you to address! I can currently move into an occupied position.
See the below image for context. Please address this REQUIRED change.**
Screen Shot 2019-10-21 at 9 41 11 PM


***Please makes changes and submit another review request.
If you have any questions, let me know here or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media or email.***

*Happy coding 😄,*
**[--Maya](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)

---
*P.S. all requested changes not marked by [OPTIONAL] or are not positive comments are **REQUIRED***

**Review by mcrd25 — APPROVED** [review]
# Milestone Approved. Unto ⏭  

Hi team,

Well done fixing the bug! Your project has been well done so far and I see no need for further review! 🙂 👍 💯 

You may now merge this branch to `development`.

***If you have any questions, now or in the future, do not hesitate to contact me on social media, email or on slack and I will be glad to chat with you.***

*Happy coding 😄,*
**[--Maya Douglas](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)

**Comment by geraldgsh** [comment]
Hi @mcrd25 thanks for taking the time to review our codes 😄

**Pull request merged**